go.mod

```diff
@@ -3,53 +3,70 @@ module github.com/kubeflow/kfserving
 	github.com/astaxie/beego v1.12.1
-	github.com/aws/aws-sdk-go v1.28.0
+	github.com/aws/aws-sdk-go v1.31.12
 	github.com/beorn7/perks v1.0.1 // indirect
 	github.com/cloudevents/sdk-go v1.2.0
 	github.com/emicklei/go-restful v2.11.0+incompatible // indirect
 	github.com/fsnotify/fsnotify v1.4.9
 	github.com/getkin/kin-openapi v0.2.0
-	github.com/go-logr/logr v0.1.0
-	github.com/go-logr/zapr v0.1.1 // indirect
-	github.com/go-openapi/spec v0.19.4
+	github.com/go-logr/logr v0.2.0
+	github.com/go-openapi/spec v0.19.6
 	github.com/gogo/protobuf v1.3.1
-	github.com/golang/groupcache v0.0.0-20191002201903-404acd9df4cc // indirect
 	github.com/golang/protobuf v1.4.2
-	github.com/google/go-cmp v0.5.0
-	github.com/google/go-containerregistry v0.0.0-20190910142231-b02d448a3705 // indirect
+	github.com/google/go-cmp v0.5.2
 	github.com/google/uuid v1.1.1
-	github.com/imdario/mergo v0.3.8 // indirect
-	github.com/json-iterator/go v1.1.8
+	github.com/json-iterator/go v1.1.10
 	github.com/mattbaird/jsonpatch v0.0.0-20171005235357-81af80346b1a // indirect
 	github.com/onsi/ginkgo v1.14.0
-	github.com/onsi/gomega v1.10.1
-	github.com/pkg/errors v0.8.1
-	github.com/prometheus/common v0.7.0 // indirect
-	github.com/prometheus/procfs v0.0.5 // indirect
+	github.com/onsi/gomega v1.10.2
+	github.com/pkg/errors v0.9.1
 	github.com/satori/go.uuid v1.2.0
 	github.com/shiena/ansicolor v0.0.0-20151119151921-a422bbe96644 // indirect
-	github.com/spf13/cobra v0.0.5
-	github.com/stretchr/testify v1.4.0
-	go.uber.org/multierr v1.2.0 // indirect
-	go.uber.org/zap v1.11.0 // indirect
-	golang.org/x/net v0.0.0-20200520004742-59133d7f0dd7
-	golang.org/x/time v0.0.0-20191023065245-6d3f0bb11be5 // indirect
-	google.golang.org/grpc v1.27.0
+	github.com/spf13/cobra v1.0.0
+	github.com/stretchr/testify v1.5.1
+	golang.org/x/net v0.0.0-20200904194848-62affa334b73
+	google.golang.org/grpc v1.31.1
 	google.golang.org/protobuf v1.25.0
-	istio.io/api v0.0.0-20191115173247-e1a1952e5b81
-	istio.io/client-go v0.0.0-20191120150049-26c62a04cdbc
+	istio.io/api v0.0.0-20200715212100-dbf5277541ef
+	istio.io/client-go v0.0.0-20201005161859-d8818315d678
 	istio.io/gogo-genproto v0.0.0-20191029161641-f7d19ec0141d // indirect
-	k8s.io/api v0.17.2
-	k8s.io/apiextensions-apiserver v0.0.0-20190918161926-8f644eb6e783
-	k8s.io/apimachinery v0.17.2
-	k8s.io/client-go v0.17.2
-	k8s.io/code-generator v0.18.6 // indirect
+	k8s.io/api v0.18.8
+	k8s.io/apiextensions-apiserver v0.18.8
+	k8s.io/apimachinery v0.18.8
+	k8s.io/client-go v11.0.1-0.20190805182717-6502b5e7b1b5+incompatible
 	k8s.io/klog v1.0.0
-	k8s.io/kube-openapi v0.0.0-20200410145947-61e04a5be9a6
-	knative.dev/pkg v0.0.0-20191217184203-cf220a867b3d
-	knative.dev/serving v0.11.0
-	sigs.k8s.io/controller-runtime v0.4.0
+	k8s.io/kube-openapi v0.0.0-20200410145947-bcb3869e6f29
+	knative.dev/pkg v0.0.0-20200922164940-4bf40ad82aab
+	knative.dev/serving v0.18.0
+	sigs.k8s.io/controller-runtime v0.6.3
```

Comment:
I upgraded controller-runtime to 0.5.0, why has it to be 0.6.3? Also @yuzisun , what compatibility level do we support? In my opinion at least k8s 1.17?

Reply:
The idea of upgrading to 0.6.3 is to get to kubernetes 1.18 - the [0.6.X tree of controller-runtime is for 1.18](https://github.com/kubernetes-sigs/controller-runtime/issues/1033#issuecomment-706562042)

Reply:
Yes I understand. So if we require 1.18 then it should be fine. Now I just upgraded to 0.5.0 to support 1.17.

Reply:
@pugangxa @ryandawsonuk I think we should support at least Kubernetes 1.16 which is the min requirement of knative 0.15 and latest knative 0.18 release bumped the min requirement to 1.17. Based on the fact that this PR passed the CI  it does actually seem to work with 1.16, not sure if there could be issues which are not covered by e2e tests.

Reply:
Thanks, then I think it should be fine and I think these libs maintain the compatibility.

Reply:
I've also been checking backward-compatibility - been running this version on a k8s 1.15 cluster running knative 0.11 and istio 1.1.6. Been able to create, call and delete and not found any problems yet.